### ISSUE TITLE
HLint fails to parse file GHC-9.0 acceps:

### ISSUE THREAD

**Issue by phadej** [ndmitchell/hlint#1230]
```haskell
module Parse where

readRational :: String -> Rational -- NB: *does* handle a leading "-"
readRational top_s
  = case top_s of
      '-' : xs -> - (read_me xs)
      xs       -> read_me xs
  where
    read_me s
      = case (do { (x,"") <- readRational__ s ; return x }) of
          [x] -> x
          []  -> error ("readRational: no parse:"        ++ top_s)
          _   -> error ("readRational: ambiguous parse:" ++ top_s)

readRational__ = undefined
```

GHC-9.0.1 compiles this fine, HLint (3.3 compiled with GHC-8.6) says:

```
% hlint Parse.hs 
Parse.hs:6:19: Error: Parse error: on input `-'
Found:
    readRational top_s
      = case top_s of
  >       '-' : xs -> - (read_me xs)
          xs       -> read_me xs
      where
  

1 hint
```

**Comment by shayne-fletcher** [comment]
The issue is that HLint is default enabling `LexicalNegation`. If you compile the above program in the presence of `{-# LANGUAGE LexicalNegation #-}` with ghc-9.0  you get the error that HLint is trying to report:
```
Parse.hs:8:19: error: parse error on input ‘-’
  |
8 |       '-' : xs -> - (read_me xs)
  |                   ^
```

One possible fix appears to be writing `'-' : xs -> -(read_me)` (remove the space following the`'-'`).

**Comment by phadej** [comment]
For the record, this is second issue in short time when HLint by default enabled flags without unclear reason "why this happens" (#1222 is another one).

I'd very much like "Don't enable any extensions" configuration option.

**Comment by ndmitchell** [comment]
It sounds like LexicalNegation is an extension that should be disabled by default? Not enabled in normal GHC, changes the way parsing works, but doesn't allow additional programs - concur @shayne-fletcher ?

**Comment by shayne-fletcher** [comment]
If judged by that criteria then yes.

**Comment by ndmitchell** [comment]
I've disabled LexicalNegation - looks like there are other diffs up for the other parts of this, so closing.
